### DISCUSSION TITLE
Using a rank file with open mpi

### DISCUSSION THREAD

**Discussion by jackmiller2003** [grimme-lab/xtb#754] [Q&A]
Hello everyone,

I was just wondering if xtb supports the use of different binding strategies for its threads. In particular, I'm trying to run several different optimisation calculations on the same node but on different cores using openMPI. As a result I might execute something like the following (where hostfile has the name of the node):

`mpirun --cpu-set 12,13,14,15,16,17 --hostfile hostfile.txt xtb/build/xtb -T 6 input.xyz`

How does the -T 6 interact with open mpi here? Is the program looking for the 6 free cores provided by mpirun and then using those? In addition, if I provided two different nodes with 48 cores and gave the program `-T 96` will it parallelise itself? 

Cheers,
Jack.

**Comment by awvwgk** [comment]
xtb does not support running with MPI. If you need to distribute your xTB calculations have a look in DFTB+ which can run xTB computations with MPI.

**Reply by jackmiller2003** [reply]
Sure that makes sense -- does that also mean that CREST cannot be parallelised across nodes? :)

**Reply by awvwgk** [reply]
CREST uses OpenMP for parallelization as well.
